models/user/avatar.go

```diff
@@ -68,11 +70,15 @@ func (u *User) AvatarLinkWithSize(size int) string {
 	useLocalAvatar := false
 	autoGenerateAvatar := false
 
-	var disableGravatar bool
-	disableGravatarSetting, _ := system_model.GetSetting(system_model.KeyPictureDisableGravatar)
-	if disableGravatarSetting != nil {
-		disableGravatar = disableGravatarSetting.GetValueBool()
-	}
+	disableGravatarStr, _ := cache.GetString(system_model.GenCacheKey(system_model.KeyPictureDisableGravatar), func() (string, error) {
+		res, err := system_model.GetSetting(system_model.KeyPictureDisableGravatar)
+		if err != nil {
+			return "", err
+		}
+		return res.SettingValue, nil
+	})
+
```

Comment:
but system_model.GetSetting(system_model.KeyPictureDisableGravatar) is:

cache.GetString(system_model.GenCacheKey(system_model.KeyPictureDisableGravatar), func() (string, error) {
		res, err := system.GetSetting(key)
		if err != nil {
			return "", err
		}
		return res.SettingValue, nil
})

What's the purpose of the double cache call?

Reply:
> 

`system_model.GetSetting` has no cache. `system_module.GetSetting` has cache. To avoid import cycle, I cannot depend on `system_module.GetSetting` here.

Reply:
If every call is supposed to be cached - we should migrate the caching code down to system_model.

See https://github.com/lunny/gitea/pull/27